libvast/src/system/accountant.cpp

```diff
@@ -207,13 +224,26 @@ struct accountant_state_impl {
       uds_sink.reset(nullptr);
     }
     if (start_uds_sink) {
-      auto s = detail::make_output_stream(cfg.uds_sink.path, cfg.uds_sink.type);
-      if (s) {
-        VAST_INFO("{} writing metrics to {}", self, cfg.uds_sink.path);
-        uds_sink = std::move(*s);
+      if (cfg.uds_sink.type == detail::socket_type::datagram) {
+        auto s = detail::uds_datagram_sender::make(cfg.uds_sink.path);
+        if (s) {
+          VAST_INFO("{} writing metrics to {}", self, cfg.uds_sink.path);
+          // uds_datagram_sink = std::move(*s);
+          uds_datagram_sink = std::make_unique<detail::uds_datagram_sender>(*s);
```

Comment:
Why do you need the `make_unique` here?

Reply:
The variable is of type `unique_ptr`, and there is no move assigment operator from `value_type`. It could be turned into an optional, but since `uds_datagram_sender` is move only we wouldn't gain anything from that.